File: rest/src/main/java/com/example/Post.java
package com.example;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.UUID;


@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "posts")
public class Post implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    UUID id;
    String title;
    String content;

    @CreationTimestamp
    @Column(name = "created_at")
    LocalDateTime createdAt;

    public static Post of(String title, String content) {
        return Post.builder().title(title).content(content).build();
    }
}

File: rest-kotlin-co/src/main/kotlin/com/example/demo/Post.kt
package com.example.demo

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.hibernate.annotations.CreationTimestamp
import java.time.LocalDateTime
import java.util.*


@Entity
@Table(name = "posts")
data class Post(
    @field:Id
    @field:GeneratedValue(strategy = GenerationType.UUID)
    var id: UUID? = null,
    var title: String? = null,
    var content: String? = null,
    @field:CreationTimestamp
    @field:Column(name = "created_at")
    var createdAt: LocalDateTime? = null
)
